packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,235 @@
+   * Internally passed from ActionList. No need to pass it explicitly
+   */
+  index?: number;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+  isDefaultSelected?: boolean;
+  intent?: 'negative';
+};
+
+type ClickHandlerType = (e: React.MouseEvent<HTMLButtonElement>) => void;
+
+const makeActionListItemClickable = (
+  clickHandler: ClickHandlerType,
+): { onPress?: StyledActionListItemProps['onPress']; onClick?: ClickHandlerType } => {
+  if (isReactNative()) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/prefer-ts-expect-error
+    // @ts-ignore: ignoring ReactNative press type for the peace of mind
+    return { onPress: clickHandler };
+  }
+
+  return {
+    onClick: clickHandler,
+  };
+};
+
+const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+  const {
+    activeIndex,
+    dropdownBaseId,
+    onOptionClick,
+    selectedIndices,
+    setShouldIgnoreBlur,
+    selectionType,
+    selectInputRef,
+  } = useDropdown();
+
+  const renderOnWebAs = props.href ? 'a' : 'button';
+  const isSelected =
+    typeof props.index === 'number'
+      ? selectedIndices.includes(props.index)
+      : props.isDefaultSelected;
+
+  return (
+    <ActionListItemContext.Provider value={{ intent: props.intent }}>
+      <StyledActionListItem
+        as={!isReactNative() ? renderOnWebAs : undefined}
+        id={`${dropdownBaseId}-${props.index}`}
+        tabIndex={-1}
+        href={props.href}
+        className={activeIndex === props.index ? 'active-focus' : ''}
+        {...makeAccessible({
+          selected: isSelected,
+          role: getActionListItemRole(props.href),
+        })}
+        {...makeActionListItemClickable((e: React.MouseEvent<HTMLButtonElement>): void => {
+          if (typeof props.index === 'number') {
+            onOptionClick(e, props.index);
+          }
+          props.onClick?.(e);
+        })}
+        onFocus={() => {
+          // We don't want to keep the browser's focus on option item. We move it to selectInput
+          if (!isReactNative()) {
+            selectInputRef.current?.focus();
+          }
+        }}
+        onMouseDown={() => {
+          setShouldIgnoreBlur(true);
+        }}
+        data-value={props.value}
+        data-index={props.index}
+        // Custom props for changes in styles
+        selectionType={selectionType}
+        hasDescription={!!props.description}
+        intent={props.intent}
+        isSelected={isSelected}
+      >
+        <Box display="flex" marginTop={props.description ? 'spacing.2' : undefined}>
+          {selectionType === 'multiple' ? (
+            // Adding aria-hidden because the listbox item in multiselect in itself explains the behaviour so announcing checkbox is unneccesary and just a nice UI tweak for us
+            <Checkbox
+              isChecked={isSelected}
+              tabIndex={-1}
+              {...makeAccessible({
+                hidden: true,
+              })}
```

Comment:
Pretty sure this makeAccessible does nothing since Checkbox doesn't accept arbitrary props.

Reply:
ops lol yes. I added a Box around it and added hidden on that. It works now. Strangely the voiceover didn't announce it even without this. probably because the focus was disabled.

Reply:
Yes should be fine anyways since tabIndex={-1} is set